extension/src/popup/views/SignTransaction/index.tsx

```diff
@@ -76,20 +76,20 @@ export const SignTransaction = () => {
 
   const {
     accountToSign: _accountToSign,
+    transaction: { _fee, _networkPassphrase },
     transactionXdr,
     domain,
     isDomainListedAllowed,
     isHttpsDomain,
     flaggedKeys,
   } = tx;
 
+  // rebuild transaction to get Transaction prototypes
   const transaction = TransactionBuilder.fromXDR(
     transactionXdr,
-    networkPassphrase,
+    _networkPassphrase,
   ) as Transaction | FeeBumpTransaction;
 
-  const { fee: _fee, networkPassphrase: _networkPassphrase } = transaction;
```

Comment:
before, we were building the xdr with the network that Freighter was configured to and then comparing the network from that Transaction with Freighter's network. It was always coming back as true, of course.

Now, we grab the network from the passed Transaction. Then we rebuild it the Transaction using the xdr and the passphrase from the old Transaction